src/tabnote.ts

```diff
@@ -351,10 +351,10 @@ export class TabNote extends StemmableNote {
             this.getStemY() - this.checkStem().getHeight() + this.getStemExtension();
 
       // Draw the Flag
-      context.save();
-      this.applyStyle(context, this.flagStyle);
-      this.flag.renderText(context, flagX, flagY);
-      context.restore();
+      this.flag.draw = (): void => {
+        this.flag.renderText(context, flagX, flagY);
+      };
+      this.flag.setContext(context).drawWithStyle();
```

Comment:
Same here. Should the flag be its own class? That way we don't have to define the draw function like this.

Reply:
done